web/app/scripts/filterbar/date-range-directive.js

```diff
@@ -2,54 +2,115 @@
+                        }
+                        if (dtRange.max) {
+                            var jsMax = moment(dtRange.max, moment.ISO_8601).toDate();
+                            scope.max = calendar.formatDate(
+                                scope.calendarOptions.dateFormat,
+                                calendar.fromJSDate(jsMax)
+                            );
+                            $('#dtMaxField').calendarsPicker('setDate', calendar.fromJSDate(jsMax));
+                        }
                         scope.isMinMaxValid();
                     }
                 });
-                scope.$watch('dtMin', function(newMin) {
-                    $('#dtMinField').datepicker('update', newMin);
-                    scope.onDtRangeChange();
-                });
-                scope.$watch('dtMax', function(newMax) {
-                    $('#dtMaxField').datepicker('update', newMax);
-                    scope.onDtRangeChange();
-                });
 
-                // On change of DT value
-                scope.onDtRangeChange = function() {
-                    dtRange.min = scope.dtMin;
-                    dtRange.max = scope.dtMax;
+                /**
+                 * Update the scope and dtRange date variables whenever a new date is selected
+                 * in the calendarPicker. This is fired by the calendar's onSelect action.
+                 * @param date {CDate} The newly selected date, as a CDate.
+                 * @param minOrMax {string} A string specifying 'min' or 'max'.
+                 */
+                function updateDate(date, minOrMax) {
+                    scope[minOrMax] = calendar.formatDate(scope.calendarOptions.dateFormat, date);
+                    var gregCDate = DateLocalization.convertToCalendar(date, 'gregorian', 'en');
+                    var gregISODate = gregCDate._calendar.toJSDate(gregCDate).toJSON();
+                    dtRange[minOrMax] = gregISODate;
                     scope.updateFilter();
-                };
+                }
+
+                /**
+                 * Sets up the calendar picker based on the current language.
+                 */
+                function configureDatePicker() {
+                    var dateConfig = DateLocalization.currentDateFormats();
+                    calendar = $.calendars.instance(dateConfig.calendar, dateConfig.language);
+                    $.calendarsPicker.setDefaults($.calendarsPicker.regionalOptions['']);
+                    scope.calendarOptions = angular.extend({
+                        calendar: calendar,
+                        dateFormat: dateConfig.formats.numeric,
+                        showAnim: '',
+                    }, $.calendarsPicker.regionalOptions[dateConfig.language]);
 
+                    // Make sure that the datepicker hides when the parent dropdown does
+                    elem.on('hide.bs.dropdown', function() {
```

Comment:
Will the `stopPropagation` strategy work even if the element doesn't exist at the time the call is made? The jQuery plugin inserts the datepicker at the end of the `body` element when its associated `input` tag gains focus, but there's no wrapper element that's available from the point of view of the directive during page load, so I'm worried that there'd be nothing to attach the handler to at that point. But maybe Angular's wrapping code takes care of that?

I like the second idea -- the DOM id reference is what was used here previously, and it seems appropriate when specifically dealing with the min or max, but doing a single call on a class is definitely cleaner for this case.